### PR TITLE
[Merged by Bors] - Improvements at Github Action: improve release names and add checksum

### PR DESCRIPTION
**Pull request by dsmello**
## Motivation
Closes issue #5230
Closes PR #5234

## Changes
Adjust the release pipeline to have better artifact names (more explicit From Windows to go-spacemesh-v1.2.4-win-amd64.zip ...)
And add the sha256 of each artifact generated and publish in this release

## Test Plan
manual testing

## TODO
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Comment by dsmello** [comment]
Did I need to add those changes at the Change log?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -77,17 +77,55 @@ jobs:
...
+          wget https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ github.ref_name }}/${{ env.OUTNAME_MAC_AMD64 }}.zip
+          wget https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ github.ref_name }}/${{ env.OUTNAME_MAC_ARM64 }}.zip
+          
+          echo "SHA256_WIN_AMD64=$(sha256sum ${{ env.OUTNAME_WIN_AMD64 }}.zip)" >> $GITHUB_ENV
+          echo "SHA256_LINUX_AMD64=$(sha256sum ${{ env.OUTNAME_LINUX_AMD64 }}.zip)" >> $GITHUB_ENV
+          echo "SHA256_LINUX_ARM64=$(sha256sum ${{ env.OUTNAME_LINUX_ARM64 }}.zip)" >> $GITHUB_ENV
+          echo "SHA256_MAC_AMD64=$(sha256sum ${{ env.OUTNAME_MAC_AMD64 }}.zip)" >> $GITHUB_ENV
+          echo "SHA256_MAC_ARM64=$(sha256sum ${{ env.OUTNAME_MAC_ARM64 }}.zip)" >> $GITHUB_ENV
+
+      - name: Generate the sha356sums yaml
```
Comment (pigmej) [inline .github/workflows/release.yml]: 256? :)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -77,17 +77,55 @@ jobs:
...
+          echo "OUTNAME_WIN_AMD64=go-spacemesh-${{ github.ref_name }}-win-amd64" >> $GITHUB_ENV
+          echo "OUTNAME_LINUX_AMD64=go-spacemesh-${{ github.ref_name }}-linux-amd64" >> $GITHUB_ENV
+          echo "OUTNAME_LINUX_ARM64=go-spacemesh-${{ github.ref_name }}-linux-arm64" >> $GITHUB_ENV
+          echo "OUTNAME_MAC_AMD64=go-spacemesh-${{ github.ref_name }}-mac-amd64" >> $GITHUB_ENV
+          echo "OUTNAME_MAC_ARM64=go-spacemesh-${{ github.ref_name }}-mac-arm64" >> $GITHUB_ENV
+
+      - name: Calculate hashsums of release files
+        shell: bash
+        run: |
+          wget https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ github.ref_name }}/${{ env.OUTNAME_WIN_AMD64 }}.zip
```
Comment (pigmej) [inline .github/workflows/release.yml]: We should really not re-download files. if the attacker has access to the bucket then he could overwrite them on the fly.

**Review by pigmej — CHANGES_REQUESTED** [review]

**Comment by fasmat** [comment]
> Did I need to add those changes at the Change log?

I don't think changes to our CI need to be mentioned in the Changelog, at least I've only ever mentioned bug fixes and new features 🙂

File: .github/workflows/release.yml (hunk 2 above)
Comment (dsmello) [reply .github/workflows/release.yml]: Fixed @pigmej

**Review by pigmej — APPROVED** [review]
I'm not sure how can we test it fully.

**Review by fasmat — CHANGES_REQUESTED** [review]
https://github.com/spacemeshos/go-spacemesh/actions/runs/6813742131/job/18529013532

`sha256sum` isn't available on every runner.

**Review by fasmat — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**